homeassistant/components/sonarr/config_flow.py

```diff
@@ -73,30 +80,87 @@ async def async_step_import(
+        self, user_input: Optional[ConfigType] = None
+    ) -> Dict[str, Any]:
+        """Confirm reauth dialog."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="reauth_confirm",
+                description_placeholders={"host": self._entry_data[CONF_HOST]},
+                data_schema=vol.Schema({}),
+                errors={},
+            )
+
+        assert self.hass
+        persistent_notification.async_dismiss(self.hass, "sonarr_reauth")
+
+        return await self.async_step_user()
+
     async def async_step_user(
         self, user_input: Optional[ConfigType] = None
     ) -> Dict[str, Any]:
         """Handle a flow initiated by the user."""
-        if user_input is None:
-            return self._show_setup_form()
+        errors = {}
 
-        if CONF_VERIFY_SSL not in user_input:
-            user_input[CONF_VERIFY_SSL] = DEFAULT_VERIFY_SSL
+        if user_input is not None:
+            if self._reauth:
+                user_input = {**self._entry_data, **user_input}
+
+            if CONF_VERIFY_SSL not in user_input:
+                user_input[CONF_VERIFY_SSL] = DEFAULT_VERIFY_SSL
+
+            try:
+                await validate_input(self.hass, user_input)
+            except SonarrAccessRestricted:
+                errors = {"base": "invalid_auth"}
+            except SonarrError:
+                errors = {"base": "cannot_connect"}
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                return self.async_abort(reason="unknown")
+            else:
+                if self._reauth:
+                    return await self._async_reauth_update_entry(
+                        self._entry_id, user_input
+                    )
+
+                return self.async_create_entry(
+                    title=user_input[CONF_HOST], data=user_input
+                )
+
+        data_schema = self._get_user_data_schema()
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(data_schema),
+            errors=errors,
+        )
 
-        try:
-            await validate_input(self.hass, user_input)
-        except SonarrAccessRestricted:
-            return self._show_setup_form({"base": "invalid_auth"})
-        except SonarrError:
-            return self._show_setup_form({"base": "cannot_connect"})
-        except Exception:  # pylint: disable=broad-except
-            _LOGGER.exception("Unexpected exception")
-            return self.async_abort(reason="unknown")
+    async def _async_reauth_update_entry(
+        self, entry_id: str, data: dict
+    ) -> Dict[str, Any]:
+        """Update existing config entry."""
+        entry = self.hass.config_entries.async_get_entry(entry_id)
+        self.hass.config_entries.async_update_entry(entry, data=data)
+        await self.hass.config_entries.async_reload(entry.entry_id)
```

Comment:
Don't we automatically reload? Or is that just for options ?

Reply:
We reload when using `_abort_if_unique_id_configured` by default if updates are passed. We don't want to use that here since we want a different abort reason.